Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.h

```diff
@@ -72,21 +72,21 @@ class ScrollingTreeScrollingNodeDelegate {
     WEBCORE_EXPORT FloatSize reachableContentsSize();
     WEBCORE_EXPORT IntPoint scrollOrigin() const;
 
-    FloatPoint currentScrollPosition() const { return m_scrollingNode.currentScrollPosition(); }
-    FloatPoint minimumScrollPosition() const { return m_scrollingNode.minimumScrollPosition(); }
-    FloatPoint maximumScrollPosition() const { return m_scrollingNode.maximumScrollPosition(); }
+    FloatPoint currentScrollPosition() const { return m_scrollingNode->currentScrollPosition(); }
+    FloatPoint minimumScrollPosition() const { return m_scrollingNode->minimumScrollPosition(); }
+    FloatPoint maximumScrollPosition() const { return m_scrollingNode->maximumScrollPosition(); }
 
-    FloatSize scrollableAreaSize() const { return m_scrollingNode.scrollableAreaSize(); }
-    FloatSize totalContentsSize() const { return m_scrollingNode.totalContentsSize(); }
+    FloatSize scrollableAreaSize() const { return m_scrollingNode->scrollableAreaSize(); }
+    FloatSize totalContentsSize() const { return m_scrollingNode->totalContentsSize(); }
 
-    bool allowsHorizontalScrolling() const { return m_scrollingNode.allowsHorizontalScrolling(); }
-    bool allowsVerticalScrolling() const { return m_scrollingNode.allowsVerticalScrolling(); }
+    bool allowsHorizontalScrolling() const { return m_scrollingNode->allowsHorizontalScrolling(); }
+    bool allowsVerticalScrolling() const { return m_scrollingNode->allowsVerticalScrolling(); }
 
-    ScrollElasticity horizontalScrollElasticity() const { return m_scrollingNode.horizontalScrollElasticity(); }
-    ScrollElasticity verticalScrollElasticity() const { return m_scrollingNode.verticalScrollElasticity(); }
+    ScrollElasticity horizontalScrollElasticity() const { return m_scrollingNode->horizontalScrollElasticity(); }
+    ScrollElasticity verticalScrollElasticity() const { return m_scrollingNode->verticalScrollElasticity(); }
 
 private:
-    ScrollingTreeScrollingNode& m_scrollingNode; // FIXME : Should use a smart pointer.
+    Ref<ScrollingTreeScrollingNode> m_scrollingNode;
```

Comment:
Normally, we'd use a CheckedRef or a WeakRef depending on what ScrollingTreeScrollingNode subclasses.

Here ScrollingTreeScrollingNode subclasses, ScrollingTreeNode which subclasses ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr so we'd have to use a ThreadSafeWeakPtr.